test/rulesets/Base/arraymath.jl

```diff
@@ -49,6 +49,43 @@
     end
 
 
+    VERSION >= v"1.6.0-DEV.1536" && @testset "muladd: $T" for T in (Float64, ComplexF64)
+        @testset "add $(typeof(z))" for z in [rand(T), rand(T,3), rand(T,3,3)] #, false]
+            dz = rand(T, fill(3, ndims(z))...)
+            @testset "matrix * matrix" begin
+                A, B = rand(T,3,3), rand(T,3,3)
+                dA, dB = rand(T,3,3), rand(T,3,3)
```

Comment:
It seems like it might be a problem that this is only ever tested on matrixes A and B both the same size

Reply:
I added a non-square test